crates/ra_ide_api/src/goto_definition.rs

```diff
@@ -334,6 +345,46 @@ mod tests {
         );
     }
 
+    #[test]
+    fn goto_definition_works_for_macro_defined_fn_with_arg() {
+        check_goto(
+            "
+            //- /lib.rs
+            macro_rules! define_fn {
+                ($name:ident) => (fn $name() {})
+            }
+
+            define_fn!(
+                foo
+            )
+
+            fn bar() {
+               <|>foo();
+            }
+            ",
+            "foo FN_DEF FileId(1) [80; 83) [80; 83)",
```

Comment:
To what line does it actually point? There are less than 80 lines in this test.

Reply:
The `TextRange` is based on characters, so it should count from start of the whole string, the len is 3 characters.

Reply:
I had actually added a `dbg` macro to print the actual output and make sure it is correct. We should definitely added use a better method to write these tests :)